Source/WebCore/html/canvas/WebGLFramebuffer.cpp

```diff
@@ -865,7 +869,12 @@ void WebGLFramebuffer::setAttachmentInternal(GCGLenum attachment, GCGLenum texTa
 
     removeAttachmentInternal(locker, attachment);
     if (texture && texture->object()) {
-        m_attachments.set(attachment, WebGLTextureAttachment::create(texture, texTarget, level, layer));
+        RefPtr<WebGLAttachment> textureAttachment = WebGLTextureAttachment::create(texture, texTarget, level, layer);
+        if (attachment == GraphicsContextGL::DEPTH_STENCIL_ATTACHMENT && context()->isWebGL2()) {
+            m_attachments.set(GraphicsContextGL::DEPTH_ATTACHMENT, textureAttachment);
+            m_attachments.set(GraphicsContextGL::STENCIL_ATTACHMENT, textureAttachment);
+        }
+        m_attachments.set(attachment, textureAttachment);
```

Comment:
This should use `WTFMove(textureAttachment)`.

Reply:
Semantically, the same `textureAttachment` reference needs to be stored under three different keys in the set. Using `WTFMove` moves it once and makes two others null.

Reply:
That's incorrect and not how move semantics work. The first two times make additional references by copying the RefPtr. Using move the third time avoids reference count churn.

It's true that moving before reusing the value in the local variable would make the value null but that is not what I am suggesting.

Reply:
I believe the fix there is to write `textureAttachment.ptr()` instead of `textureAttachment` on those lines.

Reply:
Done.